flink-yarn/src/test/java/org/apache/flink/yarn/YarnClusterDescriptorTest.java

```diff
@@ -104,9 +104,7 @@ public void testFailIfTaskSlotsHigherThanMaxVcores() throws ClusterDeploymentExc
 		clusterDescriptor.setLocalJarPath(new Path(flinkJar.getPath()));
 
 		ClusterSpecification clusterSpecification = new ClusterSpecification.ClusterSpecificationBuilder()
-			.setMasterMemoryMB(1)
-			.setTaskManagerMemoryMB(1)
-			.setNumberTaskManagers(1)
+			.setTaskManagerMemoryMB(1024)
```

Comment:
Do we really need to set the `TaskManagerMemoryMB` to 1024? Or the default value(768) is enough.

Reply:
The default is not enough. It would cause memory validation failure because the derived network memory is smaller than default min.

Reply:
I'll give it a try to change the default to 1024, see if it breaks anything.

I'm actually thinking about remove `ClusterSpecification` entirely. I've visited all the usages of this class, and it seems to me all its information can be fetched directly from configuration. However, I'd like to scope this out from this PR.

Reply:
Making `ClusterSpecificationBuilder#taskManagerMemoryMB` default 1024 turns out works well. No test broken due to this change. https://travis-ci.org/xintongsong/flink/builds/637743056
I pushed a fixup commit to this PR.